src/pools/D3MAavePool.sol

```diff
@@ -0,0 +1,216 @@
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+import "./ID3MPool.sol";
+
+interface TokenLike {
+    function approve(address, uint256) external returns (bool);
+    function transfer(address, uint256) external returns (bool);
+    function transferFrom(address, address, uint256) external returns (bool);
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface VatLike {
+    function live() external view returns (uint256);
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface D3mHubLike {
+    function vat() external view returns (address);
+}
+
+// aDai: https://etherscan.io/address/0x028171bCA77440897B824Ca71D1c56caC55b68A3
+interface ATokenLike is TokenLike {
+    function scaledBalanceOf(address) external view returns (uint256);
+    function getIncentivesController() external view returns (address);
+}
+
+// Aave Lending Pool v2: https://etherscan.io/address/0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9
+interface LendingPoolLike {
+    function deposit(address asset, uint256 amount, address onBehalfOf, uint16 referralCode) external;
+    function withdraw(address asset, uint256 amount, address to) external;
+    function getReserveNormalizedIncome(address asset) external view returns (uint256);
+    function getReserveData(address asset) external view returns (
+        uint256,    // configuration
+        uint128,    // the liquidity index. Expressed in ray
+        uint128,    // variable borrow index. Expressed in ray
+        uint128,    // the current supply rate. Expressed in ray
+        uint128,    // the current variable borrow rate. Expressed in ray
+        uint128,    // the current stable borrow rate. Expressed in ray
+        uint40,     // last updated timestamp
+        address,    // address of the adai interest bearing token
+        address,    // address of the stable debt token
+        address,    // address of the variable debt token
+        address,    // address of the interest rate strategy
+        uint8       // the id of the reserve
+    );
+}
+
+// Aave Incentives Controller: https://etherscan.io/address/0xd784927ff2f95ba542bfc824c8a8a98f3495f6b5
+interface RewardsClaimerLike {
+    function REWARD_TOKEN() external returns (address);
+    function claimRewards(address[] calldata assets, uint256 amount, address to) external returns (uint256);
+}
+
+contract D3MAavePool is ID3MPool {
+
+    mapping (address => uint256) public wards;
+    address                      public hub;
+    address                      public king;  // Who gets the rewards
+
+    VatLike            public immutable vat;
+    LendingPoolLike    public immutable pool;
+    ATokenLike         public immutable stableDebt;
+    ATokenLike         public immutable variableDebt;
+    ATokenLike         public immutable adai;
+    TokenLike          public immutable asset; // Dai
```

Comment:
I think we said we will just name it `dai` (it's so in Compound so if not let's discuss).

Reply:
0b16ef0